libs/sdk-common/src/input_parser.rs

```diff
@@ -197,6 +225,80 @@ pub async fn parse(
     Err(anyhow!("Unrecognized input type"))
 }
 
+fn get_by_key(tuple_vector: &[(&str, &str)], key: &str) -> Option<String> {
+    tuple_vector
+        .iter()
+        .find(|(k, _)| *k == key)
+        .map(|(_, v)| v.to_string())
+}
+
+fn parse_bip353_record(bip353_record: String) -> Option<String> {
+    let (_, query_part) = bip353_record.split_once("?")?;
+
+    let query_params = querystring::querify(query_part);
+
+    return get_by_key(&query_params, BOLT12_PREFIX)
+        .or_else(|| get_by_key(&query_params, LNURL_PAY_PREFIX));
+}
+
+fn validate_bip353_record(decoded: &str) -> bool {
```

Comment:
The following name is more idiomatic and lets callers more easily understand the function's behavior.

```suggestion
fn is_valid_bip353_record(decoded: &str) -> bool {
```